File: ios/ProstoVPN/Theme.swift
import SwiftUI

enum Theme {
    static let bgTop = Color(hex: 0x1A110B)
    static let bgBottom = Color(hex: 0x120C08)
    static let sheetTop = Color(hex: 0x241710)
    static let sheetBottom = Color(hex: 0x170F0A)

    static let accent = Color(hex: 0xFF5000)
    static let accentWarm = Color(hex: 0xFF711F)
    static let accentDeep = Color(hex: 0xFF4000)
    static let link = Color(hex: 0xFF6A1F)
    static let accentSoft = Color(hex: 0xFF8A50)
    static let accentHover = Color(hex: 0xFFB184)

    static let text = Color(hex: 0xEEF2FF)

    static var textSecondary: Color { text.opacity(0.45) }
    static var textMuted: Color { text.opacity(0.4) }
    static var textTertiary: Color { text.opacity(0.35) }
    static var textFaint: Color { text.opacity(0.28) }
    static var glyphOff: Color { text.opacity(0.55) }

    static var background: LinearGradient {
        LinearGradient(colors: [bgTop, bgBottom], startPoint: .top, endPoint: .bottom)
    }

    static var accentGradient: LinearGradient {
        LinearGradient(colors: [accent, accentDeep], startPoint: .topLeading, endPoint: .bottomTrailing)
    }

    static var primaryGradient: LinearGradient {
        LinearGradient(colors: [accentWarm, accentDeep], startPoint: .topLeading, endPoint: .bottomTrailing)
    }

    static var successGradient: LinearGradient {
        LinearGradient(colors: [Color(hex: 0x2EC27E), Color(hex: 0x27A06A)], startPoint: .topLeading, endPoint: .bottomTrailing)
    }

    static var destructiveGradient: LinearGradient {
        LinearGradient(colors: [Color(hex: 0xFF3B30), Color(hex: 0xD92419)], startPoint: .topLeading, endPoint: .bottomTrailing)
    }

    static var sheetGradient: LinearGradient {
        LinearGradient(colors: [sheetTop, sheetBottom], startPoint: .top, endPoint: .bottom)
    }

    static var card: Color { Color.white.opacity(0.045) }
    static var rowActive: Color { Color.white.opacity(0.05) }
    static var divider: Color { Color.white.opacity(0.06) }
    static var accentTint08: Color { accent.opacity(0.08) }
    static var accentTint10: Color { accent.opacity(0.10) }
    static var accentTint12: Color { accent.opacity(0.12) }
    static var accentTint14: Color { accent.opacity(0.14) }

    static func spring(_ duration: Double = 0.25) -> Animation {
        .timingCurve(0.3, 0.9, 0.3, 1, duration: duration)
    }
}

enum Manrope: String {
    case regular = "Manrope-Regular"
    case medium = "Manrope-Medium"
    case semibold = "Manrope-SemiBold"
    case bold = "Manrope-Bold"
    case extraBold = "Manrope-ExtraBold"

    var systemWeight: Font.Weight {
        switch self {
        case .regular: return .regular
        case .medium: return .medium
        case .semibold: return .semibold
        case .bold: return .bold
        case .extraBold: return .heavy
        }
    }

    static let isAvailable: Bool = UIFont(name: Manrope.bold.rawValue, size: 12) != nil
}

extension Font {
    static func manrope(_ size: CGFloat, _ weight: Manrope) -> Font {
        if Manrope.isAvailable {
            return .custom(weight.rawValue, size: size)
        }
        return .system(size: size, weight: weight.systemWeight)
    }
}

extension View {
    func manrope(_ size: CGFloat, _ weight: Manrope) -> some View {
        font(.manrope(size, weight))
    }
}

struct LogoImage: View {
    var body: some View {
        Image("logo")
            .resizable()
            .scaledToFit()
    }
}

extension Color {
    init(hex: UInt32) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >> 8) & 0xFF) / 255,
            blue: Double(hex & 0xFF) / 255,
            opacity: 1
        )
    }
}

func flagEmoji(countryCode: String) -> String {
    countryCode.uppercased().unicodeScalars.compactMap {
        UnicodeScalar(127397 + $0.value).map(String.init)
    }.joined()
}

File: windows/src/main/kotlin/com/prostovpn/desktop/Theme.kt
package com.prostovpn.desktop

import androidx.compose.animation.core.CubicBezierEasing
import androidx.compose.animation.core.Easing
import androidx.compose.animation.core.TweenSpec
import androidx.compose.animation.core.tween
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.platform.Font
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.sp

object Theme {
    val bgTop = Color(0xFF1A110B)
    val bgBottom = Color(0xFF120C08)
    val sheetTop = Color(0xFF241710)
    val sheetBottom = Color(0xFF170F0A)

    val accent = Color(0xFFFF5000)
    val accentWarm = Color(0xFFFF711F)
    val accentDeep = Color(0xFFFF4000)
    val link = Color(0xFFFF6A1F)
    val accentSoft = Color(0xFFFF8A50)
    val accentHover = Color(0xFFFFB184)

    val text = Color(0xFFEEF2FF)

    val textSecondary = text.copy(alpha = 0.45f)
    val textMuted = text.copy(alpha = 0.40f)
    val textTertiary = text.copy(alpha = 0.35f)
    val textFaint = text.copy(alpha = 0.28f)
    val glyphOff = text.copy(alpha = 0.55f)

    val card = Color.White.copy(alpha = 0.045f)
    val rowActive = Color.White.copy(alpha = 0.05f)
    val divider = Color.White.copy(alpha = 0.06f)
    val accentTint08 = accent.copy(alpha = 0.08f)
    val accentTint10 = accent.copy(alpha = 0.10f)
    val accentTint12 = accent.copy(alpha = 0.12f)
    val accentTint14 = accent.copy(alpha = 0.14f)

    val success = Color(0xFF2EC27E)
    val successDeep = Color(0xFF27A06A)

    val background = Brush.verticalGradient(listOf(bgTop, bgBottom))
    val accentGradient = Brush.linearGradient(listOf(accent, accentDeep))
    val primaryGradient = Brush.linearGradient(listOf(accentWarm, accentDeep))
    val successGradient = Brush.linearGradient(listOf(success, successDeep))
    val sheetGradient = Brush.verticalGradient(listOf(sheetTop, sheetBottom))

    // iOS: .timingCurve(0.3, 0.9, 0.3, 1, duration:)
    val springEasing: Easing = CubicBezierEasing(0.3f, 0.9f, 0.3f, 1f)

    fun <T> spring(durationMs: Int = 250): TweenSpec<T> = tween(durationMs, easing = springEasing)
}

val ManropeFamily = FontFamily(
    Font(resource = "fonts/manrope_regular.ttf", weight = FontWeight.Normal),
    Font(resource = "fonts/manrope_medium.ttf", weight = FontWeight.Medium),
    Font(resource = "fonts/manrope_semibold.ttf", weight = FontWeight.SemiBold),
    Font(resource = "fonts/manrope_bold.ttf", weight = FontWeight.Bold),
    Font(resource = "fonts/manrope_extrabold.ttf", weight = FontWeight.ExtraBold),
)

fun manrope(
    size: TextUnit,
    weight: FontWeight,
    color: Color = Theme.text,
    letterSpacing: TextUnit = TextUnit.Unspecified,
): TextStyle = TextStyle(
    fontFamily = ManropeFamily,
    fontSize = size,
    fontWeight = weight,
    color = color,
    letterSpacing = letterSpacing,
)

/** iOS Manrope weights → Compose */
object W {
    val regular = FontWeight.Normal
    val medium = FontWeight.Medium
    val semibold = FontWeight.SemiBold
    val bold = FontWeight.Bold
    val extraBold = FontWeight.ExtraBold
}

fun flagEmoji(countryCode: String): String =
    countryCode.uppercase().map { char ->
        String(Character.toChars(127397 + char.code))
    }.joinToString("")
